pages/index.js

```diff
@@ -45,40 +45,50 @@ class Index extends Component {
     };
   }
 
-  updateChart = () => {
-    this.setState({ retirementResults: getRetirementResults(this.state) });
+  componentDidUpdate(prevProps, prevState) {
+    const nextRetirementResults = getRetirementResults(this.state);
+    if (JSON.stringify(prevState.retirementResults) !== JSON.stringify(nextRetirementResults)) {
```

Comment:
Yes, but `retirementResults` is the output of a function, and that function doesnt fire automatically.

What I wanted to do is update `retirementResults` on every state change. `updateChart()` was doing that, but because we called it inside `handleInput`, `updatedChart()` would use the prevState (async problems).

The comparison is needed because you cant call `setState` inside `componentDidUpdate` without a condition. Otherwise it would return a infinite loop.

https://reactjs.org/docs/react-component.html#componentdidupdate
"You may call setState() immediately in componentDidUpdate() but note that it must be wrapped in a condition like in the example above, or you’ll cause an infinite loop."

Reply:
Entendido, perfeito